packages/fx-core/src/plugins/resource/appstudio/plugin.ts

```diff
@@ -1697,8 +1678,33 @@ export class AppStudioPluginImpl {
       }
     }
 
-    for (const domain of validDomains) {
-      updatedManifest.validDomains?.push(domain);
+    // This should be removed in future, the valid domains will be rendered by states
+    if (updatedManifest.validDomains?.length == 0) {
+      const validDomains: string[] = [];
+      if (tabDomain) {
+        validDomains.push(tabDomain);
```

Comment:
Previously `validDomains` is `[localhost, localhost:port]`. So `[localhost:port]` can work well enough?

Reply:
Tab works fine, but I'm not sure if it covers all the scenarios, and cannot find docs. So will keep the previous logic.